packages/store/tests/action.spec.ts

```diff
@@ -262,4 +290,153 @@ describe('Action', () => {
+            actionStatus.push(executing);
+          });
+
+          store.dispatch(new Action1());
+          store.dispatch(new Action2());
+          expect(actionStatus).toEqual([true, false]);
+        });
+
+        it('should be executing between dispatch and error', () => {
+          const actionStatus: boolean[] = [];
+
+          actions.pipe(ofActionExecuting(Action1, ErrorAction)).subscribe(executing => {
+            actionStatus.push(executing);
+          });
+
+          store.dispatch(new Action1());
+          store.dispatch(new ErrorAction());
+          expect(actionStatus).toEqual([true, false]);
+        });
+      });
+      describe('async', () => {
+        it('should be executing between dispatch and complete ', fakeAsync(() => {
+          const actionStatus: boolean[] = [];
+
+          actions.pipe(ofActionExecuting(AsyncAction1, AsyncAction2)).subscribe(executing => {
+            actionStatus.push(executing);
+          });
+
+          store.dispatch(new AsyncAction1());
+          tick(1);
+          expect(actionStatus).toEqual([]);
+          tick(1);
+          store.dispatch(new AsyncAction2());
+          tick(1);
+          expect(actionStatus).toEqual([true, false]);
+        }));
+
+        it('should be executing between dispatch and error', fakeAsync(() => {
+          const actionStatus: boolean[] = [];
+
+          actions
+            .pipe(ofActionExecuting(AsyncAction1, AsyncErrorAction))
+            .subscribe(executing => {
+              actionStatus.push(executing);
+            });
+
+          store.dispatch(new AsyncAction1());
+          tick(1);
+          expect(actionStatus).toEqual([]);
+          tick(1);
+          store.dispatch(new AsyncErrorAction());
+          tick(1);
+          expect(actionStatus).toEqual([true, false]);
+        }));
+
+        it('should be executing when action is dispatched multiple times', fakeAsync(() => {
+          const actionStatus: boolean[] = [];
+
+          actions.pipe(ofActionExecuting(AsyncAction1)).subscribe(executing => {
+            actionStatus.push(executing);
+          });
+
+          store.dispatch(new AsyncAction1());
+          expect(actionStatus).toEqual([true]);
+          store.dispatch(new AsyncAction1());
+          tick(1);
+          expect(actionStatus).toEqual([true, true, true, false]);
+        }));
+
+        it('should be executing when action is dispatched multiple times (case 2)', fakeAsync(() => {
+          const actionStatus: boolean[] = [];
+
+          actions.pipe(ofActionExecuting(AsyncAction1)).subscribe(executing => {
+            actionStatus.push(executing);
+          });
+
+          store.dispatch(new AsyncAction1());
+          store.dispatch(new AsyncAction1());
+          expect(actionStatus).toEqual([true, true]);
+          tick(1);
+          expect(actionStatus).toEqual([true, true, true, false]);
+        }));
+      });
```

Comment:
could you add a test where will

```ts
class MyState {
   @Action(AsyncAction1)
    asyncAction1(dispatch) {
      return dispatch(new AsyncAction2()).pipe(delay(0));
    }

    @Action(AsyncAction2)
    asyncAction2(dispatch) {
      return dispatch(new AsyncAction3()).pipe(delay(0));
    }

    @Action(AsyncAction3)
    asyncAction3() {
      return of({}).pipe(delay(0));
    }

}
```

Reply:
added

Reply:
very good!!!
also could you add a test where will

```ts
class MyState {
   @Action([AsyncAction1, AsyncAction2])
   combineAction(dispatch) {
      return dispatch(new AsyncAction3()).pipe(delay(0));
    }

    @Action(AsyncAction2)
    asyncAction2() {
      return of({}).pipe(delay( 100 ));
    }

    @Action(AsyncAction3)
    asyncAction3() {
      return of({}).pipe(delay(0));
    }
}

/// ......

actions.pipe(ofActionExecuting(AsyncAction1, AsyncAction2)).subscribe(executing => {
   combineStatus.push(executing);
});

actions.pipe(ofActionExecuting(AsyncAction1, AsyncAction2, AsyncAction3)).subscribe(executing => {
   actionStatus.push(executing);
});

// scenario 1
store.dispatch(new AsyncAction1());

// scenario 2
store.dispatch([ new AsyncAction1(), new AsyncAction2() ]);

// scenario 3
store.dispatch(new AsyncAction2());

// scenario 4
store.dispatch([ new AsyncAction1(), new AsyncAction2(), new AsyncAction3() ]);
```

Reply:
done, had to refactor things a bit tho